kernel/usage.c

```diff
@@ -272,6 +272,27 @@ int k_thread_runtime_stats_disable(k_tid_t  thread)
 
 	return 0;
 }
+
+bool k_thread_runtime_stats_is_enabled(k_tid_t thread)
+{
+	bool status;
+	k_spinlock_key_t key;
+
+	/*
+	 * To avoid ambiguity when return false, errno is
+	 * set to EINVAL when return false.
+	 */
+	CHECKIF(thread == NULL) {
+		errno = EINVAL;
+		return false;
+	}
+
+	key = k_spin_lock(&usage_lock);
+	status = thread->base.usage.track_usage;
+	k_spin_unlock(&usage_lock, key);
```

Comment:
Driving by: this lock is needless, remove.  You'd need to lock it if you were inspecting it and taking some action based on the value and want to be sure it's not changing while you make that decision.  But you're not doing anything but reading the value and returning it to an unsynchronized context.

Reply:
I've change it.